components/scream/src/share/io/scorpio_output.cpp

```diff
@@ -540,8 +550,14 @@ AtmosphereOutput::get_var_dof_offsets(const FieldLayout& layout)
   //       All we need to do in this routine is to compute the offset of all the entries
   //       of the MPI-local array w.r.t. the global array. So long as the offsets are in
   //       the same order as the corresponding entry in the data to be read/written, we're good.
+  // NOTE: In the case of regional output this rank may have 0 columns to write, thus, var_dof
+  //       should be empty, we check for this special case and return an empty var_dof.
   if (layout.has_tag(ShortFieldTagsNames::COL)) {
     const int num_cols = m_io_grid->get_num_local_dofs();
+    if (num_cols==0) {
+      var_dof = std::vector<scorpio::offset_t>(0);
```

Comment:
This line is not necessary. If `num_cols==0`, the layout size is 0, so the vector was already created with the correct length.